src/HexagonGrid.js

```diff
@@ -1,129 +1,134 @@
-    const x = this.props.x ? this.props.x : 0;
-    const y = this.props.y ? this.props.y : 0;
-    return (
-      <svg width={this.props.gridWidth} height={this.props.gridHeight} x={x} y={y} >
-        {
-          Array.from(Array(rows).keys()).map((row) => {
-            const remaining = this.props.hexagons.length - iHexagon;
-            const columns = remaining < this.state.columns ? remaining : this.state.columns;
-            const rowDim = this.getRowDimensions(row);
-            return (
-              <svg key={row} width={rowDim.width} height={rowDim.height} y={rowDim.y}>
-                {
-                  Array.from(Array(columns).keys()).map((col) => {
-                    const hexagon = this.props.hexagons[iHexagon];
-                    const hexDim = this.getHexDimensions(row, col);
-                    const hexProps = isFunction(this.props.hexProps) ?
-                            this.props.hexProps(hexagon) : this.props.hexProps;
-                    return (
-                      <svg
-                        key={iHexagon++}
-                        height={hexDim.height}
-                        width={hexDim.width}
-                        x={`${hexDim.x}px`}
-                      >
-                        <Hexagon {...hexProps} flatTop>
-                          {
-                            isFunction(this.props.renderHexagonContent) ?
-                              this.props.renderHexagonContent(hexagon) : <tspan />
-                          }
-                        </Hexagon>
-                      </svg>
-                    );
-                  })
-                }
-              </svg>
-            );
-          })
-        }
-      </svg>
-    );
-  }
+  return (
+    <svg width={gridWidth} height={gridHeight} x={x} y={y} >
+      {
+        Array.from(Array(state.rows).keys()).map((row) => {
+          const remaining = hexagons.length - iHexagon;
+          const columns = remaining < state.columns ? remaining : state.columns;
+          const rowDim = getRowDimensions(row);
+          return (
+            <svg key={row} width={rowDim.width} height={rowDim.height} y={rowDim.y}>
+              {
+                Array.from(Array(columns).keys()).map((col) => {
+                  const hexagon = hexagons[iHexagon];
+                  const hexDim = getHexDimensions(row, col);
+                  const _hexProps = isFunction(hexProps) ?
+                          hexProps(hexagon) : hexProps;
+                  return (
+                    <svg
+                      key={iHexagon++}
+                      height={hexDim.height}
+                      width={hexDim.width}
+                      x={`${hexDim.x}px`}
+                    >
+                      <Hexagon {..._hexProps} flatTop>
+                        {
+                          isFunction(renderHexagonContent) ?
+                            renderHexagonContent(hexagon) : <tspan />
```

Comment:
It would be good to extract this given that you're also using the same logic above.
```
function tryInvoke(func, params = [], defaultValue = null) {
  return isFunction(func)
    ? func(...params)
    : defaultValue;
}
...
tryInvoke(renderHexagonContent, [hexagon], <tspan />)
```